### DISCUSSION TITLE
Trying to read image properties in predict.py

### DISCUSSION THREAD

**Discussion by Stevew7777** [ultralytics/yolov5#9217] [General]
Ive managed to extract the bounding box xywh parameters in YOLO format from adding some code to Predict.py.

However, the current input image I'm working with is 1920 x 1280 pixels and I want to be able to work out each of the 4 bounding box corner positions ( x,y ) in relation to the original 1920 x 1280 image, so I can then work with the top right quadrant of the bounding box area when its overlayed on the original image.

I have 2 questions :

(1) If when I run predict.py the --imgsz input parameter is set at 640 x 640, will the 1920 x 1280 pixel image be left as 1920 x 1280  before being processed? Or will predict.py truncate or scale down the image to 640 x 640 before being processed?

(2) if the image is not resized to 640 x 640, how do I extract the original input image size ( e.g. 1920 x 1280 ) from within predict.py please, so I can output the bounding box corners in pixels?  Its not immediately obvious. I could import numpy and work with that but would rather not.

Any help appreciated.

-----------------------
A hint for anyone struggling with predict.py telling you you have a mismatch between class labels etc. 
Predict.py will be right, you probably have a "15" ( or whatever )  where a "0" should be in the first column in a label file somewhere  - this code will find what youre looking for quickly in linux ( this helped me, so I'm giving back to the community ...)

grep  -rIH     '<text_to_look_for>'  

how to use example :   

grep -rIH  '15'

**Comment by spacewalk01** [comment]
It will make a copy of the original image with size of 1920 x 1280 and resize it into 640x640 and run inference on 640x640 image and produce prediction boxes and rescale them into 1920 x 1280 image size.

**Reply by Stevew7777** [reply]
Thanks for the response.

And if I specified 320 x 320 size, I'm assuming it would scale all images larger than 320 x 320 to that size as well? 
I'm assuming any images smaller than 320 x 320 would scale those images up to 320 x 320? 

If you have an image thats rectangular, does it stretch the image as needed to bring it to 640 x 640?  Does this affect accuracy of detection? 

This is helpful info to know....

**Reply by spacewalk01** [reply]
https://github.com/ultralytics/yolov5/blob/5fb267f3e5dc86675d508e1b08d20fc0e2e84003/utils/dataloaders.py#L256

**Reply by spacewalk01** [reply]
https://github.com/ultralytics/yolov5/blob/5fb267f3e5dc86675d508e1b08d20fc0e2e84003/utils/augmentations.py#L135

**Reply by spacewalk01** [reply]
I think it will affect the accuracy as you lose pixel information when you downsize your image lower than your model size. It is better to leave it as default or set it to your model input size.

**Reply by Stevew7777** [reply]
OK, thank you. It was a kind of theoretical question - I  will keep images at model size like you suggest. Thanks.
